File: api/api/vfs/file.js
const map = {
  "READ_FILE": {
    args: [ "STRING" ]
  },
  "WRITE_FILE": {
    args: [ "STRING", "STRING" ]
  },
  "APPEND_FILE": {
    args: [ "STRING", "STRING" ]
  },
  "DELETE_FILE": {
    args: [ "STRING" ]
  },
  "EXISTS_FILE": {
    args: [ "STRING" ]
  }
}

const api = {

}

import { presenter } from "../presenter.js";
class File extends presenter {

  constructor(path = ""){
    super();
    this.path = path;
  }

  async delete(){
    return await this.api.deleteFile(this.path);
  }

  async exists(){
    return await this.api.existsFile(this.path);
  }

  async read(){
    return await this.api.readFile(this.path);
  }

  async write(content){
    return await this.api.writeFile(this.path, content);
  }

  async append(content){
    return await this.api.appendFile(this.path, content);
  }

}

export default { map, api, presenter: File }

File: platforms/android/contribute/src/main/java/pulse/contribute/android/api/vfs/File.kt
package pulse.contribute.android.api.vfs

import pulse.contribute.android.util.Helper
import pulse.contribute.android.util.Promise
import java.io.File
import java.lang.Exception

class File { companion object {

    fun writeFile(promiseId: String, path: String, content: String) {
        return Promise(promiseId).resolve(
            Helper.tryActionOrFalse {
                File(path).writeText(content, Charsets.UTF_8)
            }
        )
    }

    fun appendFile(promiseId: String, path: String, content: String) {
        return Promise(promiseId).resolve(
            Helper.tryActionOrFalse {
                File(path).appendText(content, Charsets.UTF_8)
            }
        )
    }

    fun readFile(promiseId: String, path: String) {
        return try {
            val file = File(path)
            Promise(promiseId).resolve(file.readText(Charsets.UTF_8))
        } catch (e: Exception) {
            Promise(promiseId).resolveUndefined()
        }
    }

    fun deleteFile(promiseId: String, path: String) {
        return try {
            val file = File(path)
            Promise(promiseId).resolve(file.exists() && file.isFile && file.delete())
        } catch (e: Exception) {
            Promise(promiseId).resolveFalse()
        }
    }

    fun existsFile(promiseId: String, path: String) {
        return try {
            val file = File(path)
            Promise(promiseId).resolve(file.exists() && file.isFile)
        } catch (e: Exception) {
            Promise(promiseId).resolve(false)
        }
    }

} }